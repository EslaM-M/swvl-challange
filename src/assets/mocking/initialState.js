// import route from "../data/route";
import tripInformation from "../data/trip-information";
export const initialState = {
  bookings: [],
  bookingAllowed: true,
  route: tripInformation.route.map(e => {
    e.count = 0;
    return e;
  }),
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
