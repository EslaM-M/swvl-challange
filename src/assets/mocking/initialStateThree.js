// import route from "../data/route";
import tripInformation from "../data/trip-information";
import fullBooking from "../data/full-booking-list";
export const initialStateThree = {
  bookings: fullBooking,
  bookingAllowed: true,
  route: tripInformation.route.map(e => {
    e.count = 0;
    return e;
  }),
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
