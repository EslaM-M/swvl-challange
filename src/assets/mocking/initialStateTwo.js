// import route from "../data/route";
import tripInformation from "../data/trip-information";
import bookings from "../data/booking";
export const initialState = {
  bookings: [],
  bookingAllowed: true,
  route: tripInformation.route,
  bookings: bookings,
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
