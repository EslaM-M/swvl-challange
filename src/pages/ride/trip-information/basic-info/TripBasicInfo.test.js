import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TripBasicInfo from "./index";
import { initialState } from "../../../../assets/mocking/initialState";
describe("TripBasicInfo Component", () => {
  let realUseContext;
  let useContextMock;
  // Setup mock
  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
    useContextMock.mockReturnValue({
      state: initialState
    });
  });
  // Cleanup mock
  afterEach(() => {
    React.useContext = realUseContext;
  });

  test("Component snapshot render correctly", () => {
    const container = render(<TripBasicInfo />);
    expect(container).toMatchSnapshot();
  });

  test("Component Render Data Correctly", async () => {
    const { getByTestId } = render(<TripBasicInfo />);
    expect(getByTestId("driver-name").innerHTML).toBe("Herbert Patton");
    expect(getByTestId("driver-rating").innerHTML).toBe("5");
    expect(getByTestId("bus-modal").innerHTML).toBe("Toyota HiAce");
    expect(getByTestId("bus-plate").innerHTML).toBe("GA 2770");
    expect(getByTestId("start-point").innerHTML).toBe("The 5th Settlement");
    expect(getByTestId("end-point").innerHTML).toBe("Almazah");
    expect(getByTestId("distance").innerHTML).toBe("0");
    expect(getByTestId("price").innerHTML).toBe("45");
    expect(getByTestId("currency").innerHTML).toBe("EGP");
  });
});
