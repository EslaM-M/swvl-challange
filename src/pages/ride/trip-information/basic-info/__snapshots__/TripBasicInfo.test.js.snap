// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripBasicInfo Component Component snapshot render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tripContent"
      >
        <div
          class="userCarDetails"
        >
          <div>
            <img
              alt="driver"
              class="avatar"
              src="user1.jpg"
            />
            <img
              alt="bus"
              class="avatar overlabAvatar"
              src="newbus.png"
            />
          </div>
          <div>
            <p>
              <label
                data-testid="driver-name"
              >
                Herbert Patton
              </label>
               
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                />
              </svg>
              <label
                data-testid="driver-rating"
              >
                5
              </label>
            </p>
            <p>
              <label
                data-testid="bus-modal"
              >
                Toyota HiAce
              </label>
               -
               
              <label
                data-testid="bus-plate"
              >
                GA 2770
              </label>
            </p>
          </div>
        </div>
        <div
          class="routeInfo"
        >
          <div
            class="leftLine"
          />
          <div>
            <p
              data-testid="start-point"
            >
              The 5th Settlement
            </p>
            <p
              data-testid="end-point"
            >
              Almazah
            </p>
          </div>
        </div>
        <div
          class="distancePrice"
        >
          <p>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 288 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M112 316.94v156.69l22.02 33.02c4.75 7.12 15.22 7.12 19.97 0L176 473.63V316.94c-10.39 1.92-21.06 3.06-32 3.06s-21.61-1.14-32-3.06zM144 0C64.47 0 0 64.47 0 144s64.47 144 144 144 144-64.47 144-144S223.53 0 144 0zm0 76c-37.5 0-68 30.5-68 68 0 6.62-5.38 12-12 12s-12-5.38-12-12c0-50.73 41.28-92 92-92 6.62 0 12 5.38 12 12s-5.38 12-12 12z"
              />
            </svg>
             Trip Distance:
             
            <label
              data-testid="distance"
            >
              0
            </label>
             
            KM
          </p>
          <p>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 288 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M209.2 233.4l-108-31.6C88.7 198.2 80 186.5 80 173.5c0-16.3 13.2-29.5 29.5-29.5h66.3c12.2 0 24.2 3.7 34.2 10.5 6.1 4.1 14.3 3.1 19.5-2l34.8-34c7.1-6.9 6.1-18.4-1.8-24.5C238 74.8 207.4 64.1 176 64V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48h-2.5C45.8 64-5.4 118.7.5 183.6c4.2 46.1 39.4 83.6 83.8 96.6l102.5 30c12.5 3.7 21.2 15.3 21.2 28.3 0 16.3-13.2 29.5-29.5 29.5h-66.3C100 368 88 364.3 78 357.5c-6.1-4.1-14.3-3.1-19.5 2l-34.8 34c-7.1 6.9-6.1 18.4 1.8 24.5 24.5 19.2 55.1 29.9 86.5 30v48c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-48.2c46.6-.9 90.3-28.6 105.7-72.7 21.5-61.6-14.6-124.8-72.5-141.7z"
              />
            </svg>
            Trip Base Fare:
            <label
              data-testid="price"
            >
              45
            </label>
            <label
              data-testid="currency"
            >
              EGP
            </label>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tripContent"
    >
      <div
        class="userCarDetails"
      >
        <div>
          <img
            alt="driver"
            class="avatar"
            src="user1.jpg"
          />
          <img
            alt="bus"
            class="avatar overlabAvatar"
            src="newbus.png"
          />
        </div>
        <div>
          <p>
            <label
              data-testid="driver-name"
            >
              Herbert Patton
            </label>
             
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
              />
            </svg>
            <label
              data-testid="driver-rating"
            >
              5
            </label>
          </p>
          <p>
            <label
              data-testid="bus-modal"
            >
              Toyota HiAce
            </label>
             -
             
            <label
              data-testid="bus-plate"
            >
              GA 2770
            </label>
          </p>
        </div>
      </div>
      <div
        class="routeInfo"
      >
        <div
          class="leftLine"
        />
        <div>
          <p
            data-testid="start-point"
          >
            The 5th Settlement
          </p>
          <p
            data-testid="end-point"
          >
            Almazah
          </p>
        </div>
      </div>
      <div
        class="distancePrice"
      >
        <p>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 288 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M112 316.94v156.69l22.02 33.02c4.75 7.12 15.22 7.12 19.97 0L176 473.63V316.94c-10.39 1.92-21.06 3.06-32 3.06s-21.61-1.14-32-3.06zM144 0C64.47 0 0 64.47 0 144s64.47 144 144 144 144-64.47 144-144S223.53 0 144 0zm0 76c-37.5 0-68 30.5-68 68 0 6.62-5.38 12-12 12s-12-5.38-12-12c0-50.73 41.28-92 92-92 6.62 0 12 5.38 12 12s-5.38 12-12 12z"
            />
          </svg>
           Trip Distance:
           
          <label
            data-testid="distance"
          >
            0
          </label>
           
          KM
        </p>
        <p>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 288 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M209.2 233.4l-108-31.6C88.7 198.2 80 186.5 80 173.5c0-16.3 13.2-29.5 29.5-29.5h66.3c12.2 0 24.2 3.7 34.2 10.5 6.1 4.1 14.3 3.1 19.5-2l34.8-34c7.1-6.9 6.1-18.4-1.8-24.5C238 74.8 207.4 64.1 176 64V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48h-2.5C45.8 64-5.4 118.7.5 183.6c4.2 46.1 39.4 83.6 83.8 96.6l102.5 30c12.5 3.7 21.2 15.3 21.2 28.3 0 16.3-13.2 29.5-29.5 29.5h-66.3C100 368 88 364.3 78 357.5c-6.1-4.1-14.3-3.1-19.5 2l-34.8 34c-7.1 6.9-6.1 18.4 1.8 24.5 24.5 19.2 55.1 29.9 86.5 30v48c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-48.2c46.6-.9 90.3-28.6 105.7-72.7 21.5-61.6-14.6-124.8-72.5-141.7z"
            />
          </svg>
          Trip Base Fare:
          <label
            data-testid="price"
          >
            45
          </label>
          <label
            data-testid="currency"
          >
            EGP
          </label>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
